### PR TITLE
GPS Sensor Fixes

### PR DESCRIPTION
**Pull request by zyunlam**

### CODE REVIEW

File: MIDAS/src/hardware/GPSSensor.cpp [hunk 1]
```diff
@@ -17,14 +17,18 @@ ErrorCode GPSSensor::init() {
...
 GPS GPSSensor::read() {
     teseo.update();
     GPGGA_Info_t gpgga_message = teseo.getGPGGAData();
     GPRMC_Info_t gprmc_message = teseo.getGPRMCData();
-    // Divide latitude and longitude by 100 because NMEA data is ddmmm.mmm or something like that
     // Also check if it's north or south
-    float lat = gpgga_message.xyz.lat / 100.f * ((gpgga_message.xyz.ns == 'N') ? 1 : -1);
-    float lon = gpgga_message.xyz.lon / 100.f * ((gpgga_message.xyz.ns == 'E') ? 1 : -1);
+    float lat = gpgga_message.xyz.lat * ((gpgga_message.xyz.ns == 'N') ? 1. : -1.);
+    float lon = gpgga_message.xyz.lon * ((gpgga_message.xyz.ns == 'E') ? 1. : -1.);
```
Comment (NPIPHI) [inline MIDAS/src/hardware/GPSSensor.cpp:27]: These number should not be floats. Lat and lon need to be doubles to avoid rounding errors

**Review by NPIPHI — CHANGES_REQUESTED** [review]
Gps coords should never end up in floats, not enough precision.

**Review by aidancostello — APPROVED** [review]

File: MIDAS/src/hardware/GPSSensor.cpp [hunk 2]
```diff
@@ -17,17 +17,22 @@ ErrorCode GPSSensor::init() {
...
+    lat = static_cast<int>(lat)/100 + std::fmod(lat, 100.)/60.;
+    lon = static_cast<int>(lon)/100 + std::fmod(lon, 100.)/60.;
+    lat *= (gpgga_message.xyz.ns == 'N') ? 1. : -1.;
+    lon *= (gpgga_message.xyz.ew == 'E') ? 1. : -1.;
     float alt = gpgga_message.xyz.alt;
     float v = gprmc_message.speed;
     uint16_t sat_count = gpgga_message.sats;
 
-    return GPS{lat, lon, alt, v, sat_count};
+    return GPS{static_cast<float>(lat), static_cast<float>(lon), alt, v, sat_count};
```
Comment (NPIPHI) [inline MIDAS/src/hardware/GPSSensor.cpp:37]: Still cannot be float at any point, floating point precision * earth circumference = hundred meter + errors

**Review by NPIPHI — CHANGES_REQUESTED** [review]
still float issues

**Review by NPIPHI — APPROVED** [review]

**Review by aidancostello — APPROVED** [review]
lgtm

**Pull request merged**